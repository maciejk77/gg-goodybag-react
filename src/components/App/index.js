// code with React Hooks

import React, { Fragment, useState } from 'react';
import cn from 'classnames';
import '../../styles/styles.scss';
import '../../styles/goodybags.scss';
import '../../../node_modules/font-awesome/css/font-awesome.min.css';

const App = () => {

  const [activeGoodybagIndex, setActiveGoodybagIndex] = useState(2);
  
  const data = { 
    goodybags: [
        { sku: "BD01", price: 500, maxData: 500, maxVoice: 300, maxTexts: 500, promoCaption: null },
        { sku: "BD21", price: 750, maxData: 2000, maxVoice: 500, maxTexts: "Unlimited", promoCaption: null},
        { sku: "BD27", price: 1000, maxData: 3000, maxVoice: "Unlimited", maxTexts: "Unlimited", promoCaption: null },
        { sku: "BD30", price: 1200, maxData: 4000, maxVoice: "Unlimited", maxTexts: "Unlimited", promoCaption: null },
        { sku: "BD33", price: 1500, maxData: 8000, maxVoice: "Unlimited", maxTexts: "Unlimited", promoCaption: null },
        { sku: "BD34", price: 2000, maxData: 20000, maxVoice: "Unlimited", maxTexts: "Unlimited", promoCaption: null },
        { sku: "BD37", price: 2500, maxData: "Always On*", maxVoice: "Unlimited", maxTexts: "Unlimited", promoCaption: "Always On* promo caption here" },
    ]
  };

  const renderGoodybag = (i) => {
    const { goodybags } = data;
  
    return goodybags.map((goodybag, index) => {
      const goodybagDataSize = goodybag.maxData > 999 ? `${goodybag.maxData / 1000} GB` : `${goodybag.maxData} MB`;
      const goodybagDataSizeDefined = isNaN(goodybag.maxData) ? goodybag.maxData : goodybagDataSize;
      
      if(i === index) { 
        return (
          <Fragment key={goodybag.sku}>
            <div className="goodybag-selector">
              <div className="goodybag__previous-button" onClick={handlePrevious}><i className="fa fa-chevron-left"></i></div>
              <div className={cn(goodybag.sku, "goodybag__sku")}></div>
              <div className="goodybag__next-button" onClick={handleNext}><i className="fa fa-chevron-right"></i></div>
            </div>
            {/* <div className="goodybag__price">Â£{goodybag.price / 100} goodybag</div> */}
            <div className="goodybag__max-data"><span className="text-bolder">{goodybagDataSizeDefined}</span> data</div>
            <div className="goodybag__max-voice"><span className="text-bolder">{goodybag.maxVoice}</span> minutes</div>
            <div className="goodybag__max-texts"><span className="text-bolder">{goodybag.maxTexts}</span> texts</div>
            { goodybag.promoCaption && 
              <div className="goodybag__promo-caption">{goodybag.promoCaption}</div> 
            }
          </Fragment>
        )
      } else { return null }
    })
  }
  
  const handleNext = () => {
    activeGoodybagIndex === data.goodybags.length - 1 
      ? setActiveGoodybagIndex(0)
      : setActiveGoodybagIndex(activeGoodybagIndex + 1)
  }


  const handlePrevious = () => {
    activeGoodybagIndex === 0
      ? setActiveGoodybagIndex(data.goodybags.length - 1)
      : setActiveGoodybagIndex(activeGoodybagIndex - 1)
  }

  return (
    <div className="goodybag">
      <div className="goodybag__text">
        {renderGoodybag(activeGoodybagIndex)}
        <div className="goodybag__all-plans">
          <div className="all-plans-text">See all plans </div>
          <i className="fa fa-caret-right"></i>
        </div>
      </div>       
    </div>
  )  
}

export default App;