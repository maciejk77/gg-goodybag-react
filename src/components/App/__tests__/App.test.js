import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../index';


describe('<App />', () => {

  // const mockData = { 
  //   goodybags: [
  //       { sku: "BD01", price: 500 },
  //       { sku: "BD21", price: 750 },
  //       { sku: "BD27", price: 1000 },
  //       { sku: "BD30", price: 1200 },
  //       { sku: "BD33", price: 1500 },
  //       { sku: "BD34", price: 2000 },
  //       { sku: "BD37", price: 2500 },
  //   ],
  //   activeGoodybagIndex: 2
  // };
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders <App />', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
  
});
