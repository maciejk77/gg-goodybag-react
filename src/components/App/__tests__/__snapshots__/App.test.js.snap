// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders <App /> 1`] = `
<div
  className="goodybag"
>
  <div
    className="goodybag__text"
  >
    <div
      className="goodybag-selector"
    >
      <div
        className="goodybag__previous-button"
        onClick={[Function]}
      >
        <i
          className="fa fa-chevron-left"
          data-cy="prev"
        />
      </div>
      <div
        className="BD27 goodybag__sku"
      />
      <div
        className="goodybag__next-button"
        onClick={[Function]}
      >
        <i
          className="fa fa-chevron-right"
          data-cy="next"
        />
      </div>
    </div>
    <div
      className="goodybag__price"
      data-cy="price"
    >
      Â£
      10
       goodybag
    </div>
    <div
      className="goodybag__max-data"
    >
      <span
        className="text-bolder"
      >
        3 GB
      </span>
       data
    </div>
    <div
      className="goodybag__max-voice"
    >
      <span
        className="text-bolder"
      >
        Unlimited
      </span>
       minutes
    </div>
    <div
      className="goodybag__max-texts"
    >
      <span
        className="text-bolder"
      >
        Unlimited
      </span>
       texts
    </div>
    <div
      className="goodybag__all-plans"
    >
      <div
        className="all-plans-text"
      >
        See all plans 
      </div>
      <i
        className="fa fa-caret-right"
      />
    </div>
  </div>
</div>
`;
